app/views/portal/show.rb

```diff
@@ -0,0 +1,871 @@
+              end
+            end
+
+            ob['text'] = text
+            subsection << ob unless text.nil? || text.blank?
+          end
+
+          if subsection.size > 0
+            section_data << subsection
+            section_labels << (section[:title].nil? ? false : t(section[:title]))
+          end
+        end
+      end
+
+      {
+        title: t(data[:title]),
+        sections: section_data.each_with_index.map do |subsection, index|
+          if subsection.size > 0
+            {
+              title: section_labels[index],
+              items: subsection
+            }
+          else
+            false
+          end
+        end
+      } unless section_data.size == 0
+    end
+
+    # def content_object_download
+    #   links = []
+
+    #   if edm_is_shown_by_download_url.present?
+    #     links << {
+    #       text: t('site.object.actions.download'),
+    #       url: edm_is_shown_by_download_url
+    #     }
+    #   end
+
+    #   return nil unless links.present?
+
+    #   {
+    #     primary: links.first,
+    #     secondary: {
+    #       items: (links.size == 1) ? nil : links[1..-1]
+    #     }
+    #   }
+    # end
+
+    def long_and_lat?
+      latitude = render_document_show_field_value(document, 'places.latitude')
+      longitude = render_document_show_field_value(document, 'places.longitude')
+      !latitude.nil? && latitude.size > 0 && !longitude.nil? && longitude.size > 0
+    end
+
+    def session_tracking_path_opts(counter)
+      {
+        per_page: params.fetch(:per_page, search_session['per_page']),
+        counter: counter,
+        search_id: current_search_session.try(:id)
+      }
+    end
+
+    def doc_title
+      # force array return with empty default
+      title = document.fetch(:title, nil)
+
+      if title.blank?
+        render_document_show_field_value(document, 'proxies.dcTitle')
+      else
+        title.first
+      end
+    end
+
+    # Media type function normalises mime types
+    # Current @mime_type variable only workd for edm_is_shown_by
+    # Once it works for web_resources we change this function so
+    # that  it accepts a mime type rather than a url, and
+
+    def media_type(url)
+      ext = url[/\.[^.]*$/]
+      if ext.nil?
+        return nil
+      end
+
+      ext = ext.downcase
+      if !['.avi', '.flac', '.mp3', '.mpga'].index(ext).nil?
+        'audio'
+      elsif !['.jpg', '.jpeg'].index(ext).nil?
+        'image'
+      elsif !['.flv', '.mp4', '.mp2', '.mpeg', '.mpg', '.ogg'].index(ext).nil?
+        'video'
+      elsif !['.txt', '.pdf'].index(ext).nil?
+        'text'
+      end
+    end
+
+    def download_disabled(rights)
+      disabled = false
+      ['http://www.europeana.eu/rights/rr-p',
+       'http://www.europeana.eu/rights/rr-r/'].map do | blacklisted |
+        if rights.index(blacklisted) == 0
+          disabled = true
+        end
+      end
+      disabled
+    end
+
+    def simple_rights_label_data(rights)
```

Comment:
Cyclomatic complexity for simple_rights_label_data is too high. [16/6]<br>Assignment Branch Condition size for simple_rights_label_data is too high. [57.01/15]<br>Method has too many lines. [90/10]<br>Perceived complexity for simple_rights_label_data is too high. [17/7]